File: haskell_euler/problem006.hs
-- Problem 6: Find the difference between the sum of the 
-- squares of the first one hundred natural numbers and 
-- the square of the sum.
-- https://projecteuler.net/problem=6

sumSquareDifference :: (Num a, Enum a) => a -> a
sumSquareDifference n = sum([1..n])^2 - sum (map (^2) [1..n])

-- Solution:
sumSquareDifference 100

File: elixir_euler/lib/problem_006.ex
# Problem 6: Sum square difference
# https://projecteuler.net/problem=6

defmodule Problem006 do
  def sum_square_difference(n) do
    sum_of_squares = Enum.map(1..n, &(&1*&1)) |> Enum.sum
    round(:math.pow(Enum.sum(1..n),2) - sum_of_squares)
  end
end
